lib/Net/Async/Blockchain/ETH.pm

```diff
@@ -335,30 +366,40 @@ async sub transform_transaction {
     my $amount        = Math::BigFloat->from_hex($decoded_transaction->{value})->bdiv(10**DEFAULT_DECIMAL_PLACES);
     my $block         = Math::BigInt->from_hex($decoded_transaction->{blockNumber});
     my $int_timestamp = Math::BigInt->from_hex($timestamp)->numify;
-
-    my $transaction;
+    my $txn_hash      = $decoded_transaction->{hash};
+    my $gas_price     = $decoded_transaction->{gasPrice};
 
     try {
-        my $gas     = $decoded_transaction->{gas};
-        my $receipt = await $self->rpc_client->get_transaction_receipt($decoded_transaction->{hash});
+        my $receipt = await $self->rpc_client->get_transaction_receipt($txn_hash);
+
+        unless ($receipt) {
+            my $flag = 1;
+            # add transaction to unprocessed transaction array so that we can process later
+            $decoded_transaction->{timestamp} = $timestamp;
+            $decoded_transaction->{flag}      = $decoded_transaction->{flag} ? $decoded_transaction->{flag} + 1 : $flag;
 
-        $gas = $receipt->{gasUsed} if $receipt && $receipt->{gasUsed};
+            # add the transaction to redis to be processed
+            await $self->redis_client->connect;
+            await $self->redis_client->rpush($redis_key => encode_json_utf8($decoded_transaction))
+                if ($decoded_transaction->{flag} && $decoded_transaction->{flag} <= 5);
 
-        # if the gas is empty we don't proceed
-        return 0 unless $gas && $decoded_transaction->{gasPrice};
+            return 0;
+        }
+
+        $gas = $receipt->{gasUsed} if $receipt->{gasUsed};
 
         # fee = gas * gasPrice
-        my $fee = Math::BigFloat->from_hex($gas)->bmul($decoded_transaction->{gasPrice});
+        $fee = Math::BigFloat->from_hex($gas)->bmul($gas_price) if $gas && $gas_price;
```

Comment:
Do we need any kind of action to take if `$fee` is `undef` or not defined? Like raising some `warn` or etc?

In other words, is it fine to set `undef` [here](https://github.com/binary-com/perl-Net-Async-Blockchain/pull/40/files#diff-cd8cd39da5fa2195c6e0eb62f9175135R402)?

Reply:
Yes, we are warning the if the fee is `undef` [here](https://github.com/regentmarkets/bom-cryptocurrency/pull/626/files#diff-de287c7b72f6c49f3b6c12c26e2ec2afR87-R90). These modules warning messages will be logged although will not be displayed in DD.. So to be aware about the logs we are warning in `bom-cryptocurrency` repo.